libcudacxx/docs/extended_api/synchronization_primitives/barrier.md

```diff
@@ -16,15 +16,30 @@ class cuda::barrier;
 
 The class template `cuda::barrier` is an extended form of [`cuda::std::barrier`]
   that takes an additional [`cuda::thread_scope`] argument.
-It has the same interface and semantics as [`cuda::std::barrier`], with the
-  following additional operations.
-
-## Barrier Operations
+If `!(scope == thread_block_scope && __isShared(this))`, then the semantics are
+the same as [`cuda::std::barrier`], with the following additional operations:
 
 | [`cuda::barrier::init`]                 | Initialize a `cuda::barrier`. `(friend function)`                       |
 | [`cuda::device::barrier_native_handle`] | Get the native handle to a `cuda::barrier`. `(function)`                |
 | [`cuda::device::barrier_arrive_tx`]     | Arrive on a `cuda::barrier` with transaction count update. `(function)` |
 
+
+Otherwise, i.e., if `scope == thread_scope_block && __isShared(this)`, then the
+semantics are that of modify
+[[thread.barrier.class]](http://eel.is/c++draft/thread.barrier.class) of ISO/IEC
+IS 14882 (the C++ Standard) as follows:
```

Comment:
The sentences reads a bit backwards, maybe
```suggestion
Otherwise, i.e., if `scope == thread_scope_block && __isShared(this)`, then the
semantics of [[thread.barrier.class]](http://eel.is/c++draft/thread.barrier.class) of ISO/IEC
IS 14882 (the C++ Standard) are modified as follows:
```